src/plugins/unified_doc_viewer/public/components/doc_viewer_table/table.tsx

```diff
@@ -265,17 +272,18 @@ export const DocViewerTable = ({
         if (!shouldShowFieldHandler(curFieldName)) {
           return acc;
         }
-
-        if (pinnedFields.includes(curFieldName)) {
+        const shouldHideNullValue = areNullValuesHidden && flattened[curFieldName] == null;
+        if (pinnedFields.includes(curFieldName) && !shouldHideNullValue) {
```

Comment:
Can we simplify like the following to avoid modifying the conditions below it?

```suggestion
        const shouldHideNullValue = areNullValuesHidden && flattened[curFieldName] == null;
        
        if (shouldHideNullValue) {
          return acc;
        }
        
        if (pinnedFields.includes(curFieldName)) {
```

Reply:
Totally, done https://github.com/elastic/kibana/pull/189601/commits/ce5d9381cc654fc110c742141d587bb1d14d6190